File: java/string/Subsequence.java
/* Author : Paras Wadhwa
 * Problem : To find if one starting is a substring of another
 * Explanation : Any string containing exact same subsequence of the string in another
 */

import java.io.*;
 
class MyCode {
  public static void main (String[] args) {
  String mainStr = "abcdebfg";
  String shortStr = "bce";
  
  int p=0;
  int count=0;
  for(int i=0;i<shortStr.length;i++){
    int j=count;
    for(j;j<mainStr.length;j++){
      if(mainStr.charAt(j)==shortStr.charAt(i)){
        p++;
        count=j;
      }
  }
  if(p==shortStr.length){
    System.out.println("true");
  }
  else{
        System.out.println("false");
		}
	}	
}
}

File: javascript/string/Subsequence.js
/* Author : Shubhankar Kotnala
 * Problem : Find if a given string is a subsequence of another string
 * Explanation : Any string whose letters come in a sequence of another is called to be its subsequence
 */
const mainStr = "abcdebfg";
const shortStr = "bce";

var p = 0;
var count = 0;
for (var i = 0; i < shortStr.length; i++) {
    var j = count;
    for (j; j < mainStr.length; j++) {
        if (mainStr.charAt(j) == shortStr.charAt(i)) {
            p++;
            count = j;
            break;
        }
    }
}
if (p === shortStr.length) {
    console.log("true");
}
else {
    console.log("false", p, shortStr.length);
}
